app/presenters/error_message/translator.rb

```diff
@@ -0,0 +1,84 @@
+#
+# This class is reponsible for retrieving plain
+# english error messages from a custom location,
+# typically error_messages.yml.
+#
+# It must retrieve english messages based on error keys
+# in various formats. These keys come from both custom validations,
+# rails validations, including nested attributes style validation
+# error messages.
+#
+
+module ErrorMessage
+  class Translator
+    include ErrorMessage::Helper
+
+    attr_reader :long_message,
+                :short_message,
+                :api_message,
+                :translations,
+                :translator
+
+    def initialize(translations, key, error)
```

Comment:
- we could accept translations
- we could have translate method on instance that returns struct/hash/ErrorMessage::Message

```
instance = ErrorMessage::Translator.new(translations)
message = instance.translate(error)
message.long
message.short
message.api
```

Reply:
May have time to get this in over the weekend

Reply:
see https://github.com/ministryofjustice/Claim-for-Crown-Court-Defence/pull/4180